Sources/SotoCore/Credential/CredentialProvider.swift

```diff
@@ -107,10 +107,10 @@ extension CredentialProviderFactory {
         }
     }
 
-    /// Use this method to load credentials from your aws cli credential file, normally located at `~/.aws/credentials`
-    public static func configFile(credentialsFilePath: String = "~/.aws/credentials", profile: String? = nil) -> CredentialProviderFactory {
+    /// Use this method to load credentials from your AWS cli credentials and optional profile configuration files, normally located at `~/.aws/credentials` and `~/.aws/config`.
+    public static func configFile(credentialsFilePath: String = "~/.aws/credentials", configFilePath: String? = nil, profile: String? = nil) -> CredentialProviderFactory {
         return Self { context in
```

Comment:
I think we should not give the option of a nil `configFilePath`. This should default to `~/.aws/config`. Also because of this I think configFilePath should be non-optional throughout the chain of function calls.

Reply:
This is fine to me, I agree with that approach. However, we will have to check for the file existence, and not error out if the file does not exist. Maybe throw an exception only if a custom patch is specified and no file is found at that path.

Reply:
Sounds good, looks like you've done this